nanomq_cli link msquic failed.

Compile nanomq with option quic 

```
cmake -G Ninja -DNNG_ENABLE_QUIC=ON ..
ninja
```
Get linker error.

```
[437/438] Linking C executable nanomq_cli/nanomq_cli
FAILED: nanomq_cli/nanomq_cli
: && /usr/bin/cc -O3 -DNDEBUG -rdynamic nanomq_cli/CMakeFiles/nanomq_cli.dir/bench.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/client.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/dds2mqtt/dds_client.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/dds2mqtt/dds_mqtt_type_conversion.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/dds2mqtt/dds_utils.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/dds2mqtt/mqtt_client.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/dds2mqtt/publisher.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/dds2mqtt/subscriber.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/dds2mqtt/vector.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/main.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/nnb_opt.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/nng_proxy.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/nngcat.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/quic_client.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/rule_ctrl.c.o nanomq_cli/CMakeFiles/nanomq_cli.dir/zmq_gateway.c.o -o nanomq_cli/nanomq_cli  -Wl,-rpath,/home/lee/workspace/nanomq/build/nng/src/supplemental/quic/msquic/msquic/bin/Release:  nng/src/supplemental/quic/msquic/msquic/bin/Release/libmsquic.so.2.2.0  nng/libnng.a  -lrt  -lpthread  -lnsl  -latomic  -ldl && :
/usr/bin/ld: nng/libnng.a(quic_api.c.o): in function `quic_open':
quic_api.c:(.text+0x25c5): undefined reference to `MsQuicOpenVersion'
/usr/bin/ld: quic_api.c:(.text+0x2680): undefined reference to `MsQuicClose'
/usr/bin/ld: nng/libnng.a(quic_api.c.o): in function `quic_close':
quic_api.c:(.text+0x2707): undefined reference to `MsQuicClose'
collect2: error: ld returned 1 exit status
[438/438] Linking C executable nanomq/nanomq
ninja: build stopped: subcommand failed.
```

--- a/nanomq_cli/CMakeLists.txt
+++ b/nanomq_cli/CMakeLists.txt
@@ -35,6 +35,8 @@ if(BUILD_DDS_PROXY)
   target_link_libraries(nanomq_cli dds_type_lib CycloneDDS::ddsc)
 endif(BUILD_DDS_PROXY)
 
+target_link_libraries(nanomq_cli nng)
+
 if(BUILD_QUIC_CLI)
   target_link_libraries(nanomq_cli msquic)
 endif(BUILD_QUIC_CLI)
@@ -60,7 +62,6 @@ endif(BUILD_VSOMEIP_GATEWAY)
 
 add_dependencies(nanomq_cli nng)
 
-target_link_libraries(nanomq_cli nng)
 
 install(TARGETS nanomq_cli RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
                                    COMPONENT Tools)

